core/src/main/java/hivemall/anomaly/SDAR2D.java

```diff
@@ -0,0 +1,166 @@
+        Preconditions.checkArgument(k >= 0, "k MUST be greather than or equals to 0: ", k);
+        Preconditions.checkArgument(k < _C.length, "k MUST be less than |C| but " + "k=" + k
+                + ", |C|=" + _C.length);
+
+        final ArrayRealVector x_t = x[0];
+        final int dims = x_t.getDimension();
+
+        if (_initialized == false) {
+            this._mu = x_t.copy();
+            this._sigma = new BlockRealMatrix(dims, dims);
+            assert (_sigma.isSquare());
+            this._initialized = true;
+            return new ArrayRealVector(dims);
+        }
+        Preconditions.checkArgument(k >= 1, "k MUST be greater than 0: ", k);
+
+        // old parameters are accessible to compute the Hellinger distance
+        this._muOld = _mu.copy();
+        this._sigmaOld = _sigma.copy();
+
+        // update mean vector
+        // \hat{mu} := (1-r) \hat{µ} + r x_t
+        this._mu = _mu.mapMultiply(1.d - _r).add(x_t.mapMultiply(_r));
+
+        // compute residuals (x - \hat{µ})
+        final RealVector[] xResidual = new RealVector[k + 1];
+        for (int j = 0; j <= k; j++) {
+            xResidual[j] = x[j].subtract(_mu);
+        }
+
+        // update covariance matrices
+        // C_j := (1-r) C_j + r (x_t - \hat{µ}) (x_{t-j} - \hat{µ})'
+        final RealMatrix[] C = this._C;
+        final RealVector rxResidual0 = xResidual[0].mapMultiply(_r); // r (x_t - \hat{µ}) 
+        for (int j = 0; j <= k; j++) {
+            RealMatrix Cj = C[j];
+            if (Cj == null) {
+                C[j] = rxResidual0.outerProduct(x[j].subtract(_mu));
+            } else {
+                C[j] = Cj.scalarMultiply(1.d - _r)
+                         .add(rxResidual0.outerProduct(x[j].subtract(_mu)));
+            }
+        }
+
+        // solve A in the following Yule-Walker equation
+        // C_j = ∑_{i=1}^{k} A_i C_{j-i} where j = 1..k, C_{-i} = C'_i 
+        /*
+         * /C_1\     /A_1\  /C_0     |C_1     |C_2     | .  .  .   |C_{k-1} \ 
+         * |---|     |---|  |--------+--------+--------+           +--------|
+         * |C_2|     |A_2|  |C_1'    |C_0     |C_1     |               .    |
+         * |---|     |---|  |--------+--------+--------+               .    |
+         * |C_3|  =  |A_3|  |C_2'    |C_1'    |C_0     |               .    |
+         * | . |     | . |  |--------+--------+--------+                    |
+         * | . |     | . |  |   .                            .              |
+         * | . |     | . |  |   .                            .              |
+         * |---|     |---|  |--------+                              +-------|
+         * \C_n/     \A_n/  \C_{k-1}'| .  .  .                      |C_0    / 
+         */
+        RealMatrix[][] rhs = MatrixUtils.toeplitz(C, k);
+        RealMatrix[] lhs = Arrays.copyOfRange(C, 1, k + 1);
+        RealMatrix R = MatrixUtils.flatten(rhs);
```

Comment:
@L3Sota Are you expecting each `C_i` as a submatrix and flatting a matrix of matrix to a matrix?

`R` will not become a Toeplitz Matrix then and it's hard to solve `A` by LUDecomposition.